src/Management/src/Endpoint/Metrics/Observer/EventCounterListener.cs

```diff
@@ -120,27 +120,22 @@ private void ExtractAndRecordMetric(string eventSourceName, IDictionary<string,
         long? longValue = null;
         string counterName = string.Empty;
         var labelSet = new List<KeyValuePair<string, object>>();
-        bool excludedMetric = false;
         string counterDisplayUnit = null;
         string counterDisplayName = null;
 
         foreach (KeyValuePair<string, object> payload in eventPayload)
         {
-            if (excludedMetric)
-            {
-                break;
-            }
-
             string key = payload.Key;
 
             switch (key)
             {
                 case var _ when key.Equals("Name", StringComparison.OrdinalIgnoreCase):
                     counterName = payload.Value.ToString();
 
-                    if (_options.ExcludedMetrics.Contains(counterName))
+                    // Inclusion list is intended to be evaluated before exclusion list
```

Comment:
```suggestion
```

Reply:
As mentioned earlier, please don't create and commit your own suggestions. Because it auto-closes the conversation, which makes it hard to track for the conversation initiator to verify that the feedback was addressed as intended.